modules/calib3d/src/solvepnp.cpp

```diff
@@ -311,19 +311,24 @@ bool solvePnPRansac(InputArray _opoints, InputArray _ipoints,
 
     opoints_inliers.resize(npoints1);
     ipoints_inliers.resize(npoints1);
-    result = solvePnP(opoints_inliers, ipoints_inliers, cameraMatrix,
-                      distCoeffs, rvec, tvec, useExtrinsicGuess,
-                      (flags == SOLVEPNP_P3P || flags == SOLVEPNP_AP3P) ? SOLVEPNP_EPNP : flags) ? 1 : -1;
+    try
+    {
+        result = solvePnP(opoints_inliers, ipoints_inliers, cameraMatrix,
+                          distCoeffs, rvec, tvec, useExtrinsicGuess,
+                          (flags == SOLVEPNP_P3P || flags == SOLVEPNP_AP3P) ? SOLVEPNP_EPNP : flags) ? 1 : -1;
+    }
+    catch (const cv::Exception& e)
+    {
+        result = 0;
+        CV_UNUSED(e);
+        CV_LOG_DEBUG(NULL, "solvePnPRansac(): solvePnP stage to compute the final pose using points "
+            "in the consensus set failed, use result from MSS (Minimal Sample Sets) stage instead:\n" << e.what());
+    }
 
-    if( result <= 0 )
+    if(result <= 0)
     {
         _rvec.assign(_local_model.col(0));    // output rotation vector
         _tvec.assign(_local_model.col(1));    // output translation vector
-
-        if( _inliers.needed() )
-            _inliers.release();
-
-        return false;
```

Comment:
This part changes behavior of the case which is not covered by PR's description:
- `result = solvePnP()` doesn't throw exception
- it returns `result <= 0`

Alternative #19492 preserves behavior of non-exception case.

---

> When I call solvePnPRansac with 69 points (data I provided above), then I should get a result (even if it's false) rather than an Exception.

`false` code path is changed.

> solvePnPRansac returns true even if the solvePnP stage failed. In this case, result from MSS stage is taken, and the exception content from solvePnP is put into log.

Only exception's case is logged, but returned value is forced "true" for other cases too.

Reply:
You are right, the logging strategy here is problematic.

Reply:
Main question is about the final returned value.

|result = solvePnP|existed code|This PR| PR 19492|case4(no PR)
|---|---|---|---|---|
|result > 0|true|true|true|true|
|result<=0|false|true (&#x2753;) |false|false|
|exception|exception|true (&#x2753;)|true (&#x2753;)| false|

Reply:
I think you meant to @ someone else (not me)?